index.js

```diff
@@ -16,12 +16,10 @@ module.exports = function (data, options) {
 			return;
 		}
 
-		if (file.data) {
-			data = _.merge(file.data, data);
-		}
+		var data = getData(file);
 
 		try {
-			file.contents = new Buffer(template(file.contents.toString(), data || {}, options));
+			file.contents = new Buffer(template(file.contents.toString(), data, options).toString());
```

Comment:
The non-precompile method didn't call `.toString()` on the `template`. Not sure if that has any effect or not though.